src/app/voting/voting-service.js

```diff
@@ -145,5 +146,16 @@ function VotingService($http, $rootScope, $q, SongFactory, ReceivedVoteFactory,
         }
     }
 
+    this.setVotesongsOfQueue = function(songs) {
+        _votesongsInQueue = {};
+        _.each(songs, function(song){
+            _votesongsInQueue[song._id] = true;
+        })
+    }
+
+    this.isSongInQueue = function(song) {
+        return _votesongsInQueue[song._id] !== undefined ? true : false;
```

Comment:
info: this may be a potential bug. if you set `_votesongsInQueue[song._id]` to `true` (for a reason), then this function will return `false` instead of `true`.
It would be better to write:
`return _votesongsInQueue[song._id] !== undefined && _votesongsInQueue[song._id] !== false ? true : false;` 
or
`return _votesongsInQueue[song._id] === true ? true : false;` 
or
`return !!_votesongsInQueue[song._id];`
_(not tested)_